### ISSUE TITLE
Cannot find type definition file for 'rollup-plugin-import-css' - TypeScript 5.1.6

### ISSUE THREAD

**Issue by forgetso** [jleeson/rollup-plugin-import-css#16]
In order to use this plugin with TypeScript 5.1.6, `rollup.d.ts` needs to be created alongside the `tsconfig.json` containing

```
declare module 'rollup-plugin-import-css'
```

Adding `rollup-plugin-import-css` to the types section in `tsconfig.ts` does not work.

```
error TS2688: Cannot find type definition file for 'rollup-plugin-import-css'.
  The file is in the program because:
    Entry point of type library 'rollup-plugin-import-css' specified in compilerOptions

  tsconfig.json:6:34
    6         "types": ["vite/client", "rollup-plugin-import-css"]
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~
    File is entry point of type library specified here.

```

Adding `types` folder to path also doesn't work.

```
error TS2688: Cannot find type definition file for 'rollup-plugin-import-css/types'.
  The file is in the program because:
    Entry point of type library 'rollup-plugin-import-css/types' specified in compilerOptions

  tsconfig.json:6:34
    6         "types": ["vite/client", "rollup-plugin-import-css/types"]
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    File is entry point of type library specified here.

```

And so on.

Reason for this bug:

```
 There are types at '/home/user/project/node_modules/rollup-plugin-import-css/types/plugin.d.ts', but this result could not be resolved when respecting package.json "exports". The 'rollup-plugin-import-css' library may need to update its package.json or typings.
```

Changelog entry:

https://devblogs.microsoft.com/typescript/announcing-typescript-5-1/#explicit-typeroots-disables-upward-walks-for-node_modules-types

Thanks!

**Comment by jleeson** [comment]
Hey @forgetso, thanks for creating this issue. It should now be resolved in v3.3.2 of this plugin. If there are still any remaining issues with resolution of types, let me know and I will re-open this issue.

**Comment by goastler** [comment]
Hi, I've bumped our version to v3.3.3 (I work for @forgetso ) and getting this error now: 
```
This expression is not callable.
  Type 'typeof import("/home/geopro/bench/captcha3/node_modules/rollup-plugin-import-css/types/plugin")' has no call signatures.
```

The code is 
```

import css from 'rollup-plugin-import-css'
...

export default async function (
    ...
): Promise<UserConfig> {
    ...
    return {
                plugins: [
                    css(), // <-- here!
                    ...
               ]
...
}
```

**Comment by goastler** [comment]
```
npm list | grep import-css
│ ├── rollup-plugin-import-css@3.3.3
│ ├── rollup-plugin-import-css@3.3.3 deduped
```

**Comment by goastler** [comment]
`css.default()` seems to work

**Comment by jleeson** [comment]
Hi @goastler, are you able to provided your full rollup config? css.default will work as its referencing the default export however you shouldn't need to. It may be trying to import the cjs version of the plugin. Knowing how you have rollup setup will enable me to determine if there is something I can change in this package to remove the requirement of using .default in cjs

**Comment by forgetso** [comment]
Hi @jleeson you can see your plugin being used [here](https://github.com/prosopo/captcha/blob/e513d332c21e01c8cfb80d614b36518cef9cb26d/dev/config/src/vite/vite.backend.config.ts#L163) in our repo.

To replicate the issue, clone the repo and try to build everything after deleting [this](https://github.com/prosopo/captcha/blob/main/dev/config/src/rollup.d.ts) file.

Thanks!

**Comment by jleeson** [comment]
@forgetso I played around with it and resolved the issue. for some reason typescript will only recognize a module exists if there is a type definition file with a matching extension. Given how this plugin is written it was only resolving the commonjs entry point. This plugin now correctly tells typescript to use the esm entrypoint when possible. I have already tested this with your repository and it correctly resolves the right bundle along with the type definitions.

Make sure you update to v3.3.4 and let me know if there are any more issues.
